org.adempiere.base/src/org/compiere/model/MProductionLineMA.java

```diff
@@ -99,17 +100,24 @@ public static MProductionLineMA get( MProductionLine parent, int asi, Timestamp
 	@Override
 	protected boolean beforeSave(boolean newRecord) 
 	{
-		MProductionLine parentLine = new MProductionLine(getCtx(), getM_ProductionLine_ID(), get_TrxName());
-		MProduction prodParent;
-		if (parentLine.getM_Production_ID() > 0) {
-			prodParent = new MProduction(getCtx(), parentLine.getM_Production_ID(), get_TrxName());
-		} else {
-			MProductionPlan plan = new MProductionPlan(getCtx(), parentLine.getM_ProductionPlan_ID(), get_TrxName());
-			prodParent = new MProduction(getCtx(), plan.getM_Production_ID(), get_TrxName());
-		}
-		if (newRecord && prodParent.isProcessed()) {
-			log.saveError("ParentComplete", Msg.translate(getCtx(), "M_Production_ID"));
-			return false;
+		if(newRecord) {

```

Comment:
This is changing the logic of beforeSave - not validating for update records, just insert, I think that's wrong

Reply:
Here I'm preserving the original logic that performed this validation upon record creation (not update). Do you want to change this? Just to confirm.

Reply:
Ah OK - my fault - didn't notice the newRecord below, you're totally right that moving the "f (newRecord" is an improvement, but I think is better to preserve the actual logic instead of replacing it with SQL

This way:
```java
	@Override
	protected boolean beforeSave(boolean newRecord) 
	{
		if (newRecord) {
			MProductionLine parentLine = new MProductionLine(getCtx(), getM_ProductionLine_ID(), get_TrxName());
			MProduction prodParent;
			if (parentLine.getM_Production_ID() > 0) {
				prodParent = new MProduction(getCtx(), parentLine.getM_Production_ID(), get_TrxName());
			} else {
				MProductionPlan plan = new MProductionPlan(getCtx(), parentLine.getM_ProductionPlan_ID(), get_TrxName());
				prodParent = new MProduction(getCtx(), plan.getM_Production_ID(), get_TrxName());
			}
			if (prodParent.isProcessed()) {
				log.saveError("ParentComplete", Msg.translate(getCtx(), "M_Production_ID"));
				return false;
			}
		}
		return true;
	}
```

Reply:
Modified code to perform this check upon record creation AND update. Keeping the rest unmodified due to reasons detailed earlier.